master/internal/user/postgres_users.go

```diff
@@ -29,11 +29,26 @@ func StartSession(ctx context.Context, user *model.User) (string, error) {
 		Expiry: time.Now().Add(SessionDuration),
 	}
 
-	_, err := db.Bun().NewInsert().
-		Model(userSession).
-		Column("user_id", "expiry").
-		Returning("id").
-		Exec(ctx, &userSession.ID)
+	err := db.Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		_, err := db.Bun().NewInsert().
+			Model(userSession).
+			Column("user_id", "expiry").
+			Returning("id").
+			Exec(ctx, &userSession.ID)
+		if err != nil {
+			return err
+		}
+
+		_, err = db.Bun().NewUpdate().
+			Table("users").
+			SetColumn("last_login", "NOW()").
```

Comment:
I've noticed that some things like this are getting added as [triggers](https://github.com/determined-ai/determined/pull/7949) and some like this. Is there any decision criteria? Should we have a stance?

Reply:
good question! I think a good heuristic for when to use one over the other is if there are multiple places the instigating change could happen -- this would cover something like `modified_at` on users as well as `best_trial_id` on experiments, which run on any updates to the `users` or `trials` tables, respectively. There's also a trigger on `user_profile_images` which updates the user's `modified_at` on change or deletion, and i think that might be a case of extending the pattern w/r/t updates to `user.modified_at`, but I'm not sure.

Reply:
Devil's advocate: There are multiple places instigating changes in `user_session`s (saml, oidc, det users), it is just one of the few places in det where all those interactions go through a well-defined interface ("StartSession"). So I think there is an argument that triggers can just be used as a bandaid for a poorly abstracted code base.

Reply:
I think my stance is that I agree with it for stuff as simple as updated_at, created_at, but after that I think we probably shouldn't use them (e.g., I think best_trial_id is too far, I'd rather us have a better abstraction around interacting with trials in the DB and have that handle it).